monitoring/draft-eranm-monitoring-api.md

```diff
@@ -0,0 +1,205 @@
+    from the corresponding certificate as described in
+    reconstructing_tbscertificate).
+: log_id:
+  : Optional, An OID of the log the query refers to.
+
+Outputs:
+
+: log_entries:
+  : An array of objects, each consisting of
+    : log_id:
+      : OID of the origin log for this entry.
+    :entries
+      : An array of 0-based indices of log entries corresponding to the supplied
+    HASH.
+
+Error codes:
+
+|------------+--------------------------------------------------------------------|
+| Error Code | Meaning                                                            |
+|------------+--------------------------------------------------------------------|
+| bad hash   | `hash` is not the right size or format.                            |
+| not found  | `sct` does not correspond to an entry that is currently available. |
+|------------+--------------------------------------------------------------------|
+
+Note that it is possible for a certificate to be logged more than once. If that
+is the case, the monitor SHALL return all the indices it knows about for this
+entry. If the certificate is present in the log, then the monitor MUST return at
+least one entry index.
+
+If the log_id input parameter is ommitted, the log SHALL return entries from all
+the logs it currently monitors.
+
+## Retrieve Signed Tree Heads between Two Times
+GET https://\<server>/ct/v2/get-sths
+
+Inputs:
+
+: start:
+  : an earlier NTP Time [RFC5905], measured in milliseconds since the epoch (January 1, 1970, 00:00 UTC), ignoring leap seconds.
+
+: end:
+  : a later NTP Time [RFC5905], measured in milliseconds since the epoch (January 1, 1970, 00:00 UTC), ignoring leap seconds.
+
+: log_id:
+  : OID of the log this request refers to.
+
+Outputs:
+
+: log_sths:
+  : An array of objects, each consisting of
+    : log_id : OID of the log which issued the STHs.
+    : sths: : an array of base64 encoded TransItem structures of type signed_tree_head_v2, signed by this log.
+
+The start and end parameters SHOULD be within the range 0 <= x < timestamp as returned by get-sth.
+
+The start parameter MUST be less than or equal to the end parameter.
+
+Servers MUST honor requests where 0 <= start < timestamp and end >= timestamp by returning a partial response covering only the STHs in the specified range. end >= timestamp could be caused by skew. Note that the following restriction may also apply:
+
+Servers MAY restrict the number of STHs that can be retrieved per get-sths request. If there are more than the permitted number of STHs in the specified range, the log SHALL return the maximum number of STHs permissible. These STHs SHALL be ordered chronologically by timestamp, oldest first, beginning with the earliest STH in the specified range.
+
+It is possible the server will not have any STHs between start and end. In this case it MUST return an empty sths array.
+
+When implemented by a log, the log_id parameter SHALL be ommitted by clients and ignored by the log.
```

Comment:
What does it mean for a subset of APIs that are part of the monitoring role to be implemented by a log server?

I'm a bit concerned that this blurring of boundaries is going to make writing interoperable tools harder rather than easier.